ironfish/src/rpc/routes/wallet/createTransaction.ts

```diff
@@ -0,0 +1,250 @@
+    feePriorityLevel: yup.string().optional(),
+    expiration: yup.number().optional(),
+    expirationDelta: yup.number().optional(),
+  })
+  .defined()
+
+export const CreateTransactionResponseSchema: yup.ObjectSchema<CreateTransactionResponse> = yup
+  .object({
+    transaction: yup.string().defined(),
+  })
+  .defined()
+
+router.register<typeof CreateTransactionRequestSchema, CreateTransactionResponse>(
+  `${ApiNamespace.wallet}/createTransaction`,
+  CreateTransactionRequestSchema,
+  async (request, node): Promise<void> => {
+    const data = request.data
+
+    const account = node.wallet.getAccountByName(data.sender)
+
+    if (!account) {
+      throw new ValidationError(`No account found with name ${data.sender}`)
+    }
+
+    // The node must be connected to the network first
+    if (!node.peerNetwork.isReady) {
+      throw new ValidationError(
+        `Your node must be connected to the Iron Fish network to send a transaction`,
+      )
+    }
+
+    if (!node.chain.synced) {
+      throw new ValidationError(
+        `Your node must be synced with the Iron Fish network to send a transaction. Please try again later`,
+      )
+    }
+
+    const totalByAssetIdentifier = new BufferMap<bigint>()
+    if (data.fee) {
+      const fee = CurrencyUtils.decode(data.fee)
+      if (fee < 1n) {
+        throw new ValidationError(`Invalid transaction fee, ${data.fee}`)
+      }
+
+      totalByAssetIdentifier.set(Asset.nativeId(), fee)
+    }
+
+    const receives = data.receives.map((receive) => {
+      let assetId = Asset.nativeId()
+      if (receive.assetId) {
+        assetId = Buffer.from(receive.assetId, 'hex')
+      }
+
+      const amount = CurrencyUtils.decode(receive.amount)
+      if (amount <= 0) {
+        throw new ValidationError(`Invalid transaction amount ${amount}.`)
+      }
+
+      const sum = totalByAssetIdentifier.get(assetId) ?? 0n
+      totalByAssetIdentifier.set(assetId, sum + amount)
+
+      return {
+        publicAddress: receive.publicAddress,
+        amount: amount,
+        memo: receive.memo,
+        assetId,
+      }
+    })
+
+    const mints: MintDescription[] = []
+    if (data.mints) {
+      for (const mint of data.mints) {
+        let asset: Asset
+        if (mint.assetId) {
+          const record = await node.chain.getAssetById(Buffer.from(mint.assetId, 'hex'))
+          if (!record) {
+            throw new ValidationError(
+              `Asset not found. Cannot mint for identifier '${mint.assetId}'`,
+            )
+          }
+
+          asset = new Asset(
+            account.spendingKey,
+            record.name.toString('utf8'),
+            record.metadata.toString('utf8'),
+          )
+          // Verify the stored asset produces the same identfier before building a transaction
+          if (!asset.id().equals(Buffer.from(mint.assetId, 'hex'))) {
+            throw new ValidationError(`Unauthorized to mint for asset '${mint.assetId}'`)
+          }
```

Comment:
We don't do this for mint, I don't think this is necessary if Rohan didn't do it. The TX won't validate.

https://github.com/iron-fish/ironfish/blob/master/ironfish/src/rpc/routes/wallet/mintAsset.ts#L74-L81

Reply:
@rohanjadvani did this in wallet.mint(), I think we should either move that logic to wallet.creationTransaction() or I will have to keep it here.

Reply:
Yeah, this existed in `wallet.mint`. It's a sanity check before it hits the circuits, so I think it's useful to have. I think we should include it in the wallet instead of RPC.

Reply:
I will move it to wallet.createTransaction from wallet.mint()